truth_set/java/structure/organisation/consistency/Supermarket.java

```diff
@@ -0,0 +1,61 @@
+import java.util.Map;
+import java.util.HashMap;
+
+class UserService {
+    public String fetchUserProfile(int userId) {
+        try {
+            Thread.sleep(1000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        return "User Profile for ID: " + userId;
+    }
+
+    public boolean checkUserEligibility(int age) {
+        return age >= 18;
+    }
+}
+
+class ProductService {
+    public String getProduct(int productId) {
+        Map<Integer, String> products = new HashMap<>();
+        products.put(101, "Laptop");
+        products.put(102, "Smartphone"); 
+        products.put(103, "Headphones");
+        return products.getOrDefault(productId, "Product Not Found");
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider optimizing data initialization approach</b></div>
<div id="fix">

Consider extracting the hardcoded data initialization into a separate method or using a configuration file. The current implementation recreates the data maps on every method call which could be inefficient.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -19,43 +19,47 @@
 class ProductService {
     private final Map<Integer, String> products;

     public ProductService() {
         products = new HashMap<>();
         products.put(101, "Laptop");
         products.put(102, "Smartphone");
         products.put(103, "Headphones");
     }

     public String getProduct(int productId) {
         return products.getOrDefault(productId, "Product Not Found");
     }
 }

 class OrderService {
     private final Map<Integer, String> orders;

     public OrderService() {
         orders = new HashMap<>();
         orders.put(5001, "Order #5001: Shipped");
         orders.put(5002, "Order #5002: Processing");
         orders.put(5003, "Order #5003: Delivered");
     }

     public String retrieveOrder(int orderId) {
         return orders.getOrDefault(orderId, "Order Not Found");
     }
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/3#issuecomment-2534994613>#cce4ef</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged